src/Core/Backup/Listing/BackupGridDataProvider.php

```diff
@@ -0,0 +1,157 @@
+
+/**
+ * Class BackupGridDataProvider provides backups for listing in grid
+ */
+final class BackupGridDataProvider implements GridDataProviderInterface
+{
+    /**
+     * @var BackupRepositoryInterface
+     */
+    private $backupRepository;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var string
+     */
+    private $languageDateTimeFormat;
+
+    /**
+     * @var BackupComparatorInterface
+     */
+    private $backupByDateComparator;
+
+    /**
+     * @param BackupRepositoryInterface $backupRepository
+     * @param BackupComparatorInterface $backupByDateComparator
+     * @param TranslatorInterface $translator
+     * @param string $languageDateTimeFormat
+     */
+    public function __construct(
+        BackupRepositoryInterface $backupRepository,
+        BackupComparatorInterface $backupByDateComparator,
+        TranslatorInterface $translator,
+        $languageDateTimeFormat
+    ) {
+        $this->backupRepository = $backupRepository;
+        $this->translator = $translator;
+        $this->languageDateTimeFormat = $languageDateTimeFormat;
+        $this->backupByDateComparator = $backupByDateComparator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getData(SearchCriteriaInterface $searchCriteria)
+    {
+        $backups = $this->backupRepository->get()->all();
+        usort($backups, [$this->backupByDateComparator, 'compare']);
+
+        $paginatedBackups = null !== $searchCriteria->getOffset() && null !== $searchCriteria->getLimit() ?
+            array_slice($backups, $searchCriteria->getOffset(), $searchCriteria->getLimit()) :
+            $backups;
+
+        $backupsArray = [];
+
+        foreach ($paginatedBackups as $backup) {
+            $backupsArray[] = [
+                'file_name' => $backup->getFileName(),
+                'file_size' => $backup->getSize(),
+                'date' => $backup->getDate(),
+                'age' => $backup->getAge(),
+                'age_formatted' => $this->getFormattedAge($backup),
+                'date_formatted' => date($this->languageDateTimeFormat, $backup->getDate()->getTimestamp()),
+                'file_size_formatted' => $this->getFormattedSize($backup),
+            ];
+        }
+
+        $backupCollection = new RecordCollection($backupsArray);
+
+        return new GridData(
+            $backupCollection,
+            count($backups)
+        );
+    }
+
+    /**
+     * Get formatted age
+     *
+     * @param BackupInterface $backup
+     *
+     * @return string
+     */
+    private function getFormattedAge(BackupInterface $backup)
+    {
+        if (TimeDefinition::HOUR_IN_SECONDS > $backup->getAge()) {
+            return sprintf('< 1 %s', $this->translator->trans('Hour', [], 'Admin.Global'));
+        }
+
+        if (TimeDefinition::DAY_IN_SECONDS > $backup->getAge()) {
+            $hours = (int) floor($backup->getAge() / TimeDefinition::HOUR_IN_SECONDS);
```

Comment:
Is casting really necessary?

Reply:
yeah, `floor()` returns `float` and one line below there's strict comparison. We can of course update it not to use casting, but is that necessary?

Reply:
You used sprintf try with `%d` instead of `%s` :)

Reply:
for printing, yup. :+1:  but i still need to use it in strict comparison (`int === int `) or switch to `int == float`, which is not good practice i guess?

Reply:
You can type ` === 1.0` it prevents a useless cast. As you wish =)